nixos/modules/services/networking/pixiecore.nix

```diff
@@ -0,0 +1,109 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.pixiecore;
+
+in
+
+{
+
+  options = {
+
+    services.pixiecore = {
+
+      enable = mkEnableOption "Pixiecore";
+
+      debug = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Log more things that aren't directly related to booting a recognized client";
+      };
+
+      logTimestamps = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Add a timestamp to each log line";
+      };
+
+      dhcpNoBind = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Handle DHCP traffic without binding to the DHCP server port";
+      };
+
+      bootMsg = mkOption {
+        type = types.nullOr types.string;
+        default = "";
+        description = "Message to print on machines before booting";
+      };
+
+      kernel = mkOption {
+        type = types.string;
+        default = "";
+        description = "Kernel path";
+      };
+
+      initrd = mkOption {
+        type = types.string;
+        default = "";
+        description = "Initrd path";
+      };
+
+      cmdLine = mkOption {
+        type = types.nullOr types.string;
+        default = "";
+        description = "Kernel commandline arguments";
+      };
+
+      listen = mkOption {
+        type = types.string;
+        default = "0.0.0.0";
+        description = "IPv4 address to listen on";
+      };
+
+      port = mkOption {
+        type = types.int;
+        default = 80;
+        description = "Port to listen on for HTTP";
+      };
+
+      statusPort = mkOption {
+        type = types.int;
+        default = 80;
```

Comment:
This is a privileged port so you'll need to use capabilities of you wish to bind to it. I believe you're looking for cap_net_bind_service. Or just use an unprivileged port.

Reply:
Will Add   AmbientCapabilities = "cap_net_bind_service"; Should keep using 80 as it's pixiecore's own default port.